### PR TITLE
[Astro] Added German localization

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added German translations - not yet complete
- Fixed typos
- Fixed linter errors

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.astro/ESH-INF/thing/sun.xml [hunk 1]
```diff
@@ -30,27 +30,27 @@
 				<description>The morning night event range</description>
 			</channel-group>
 			<channel-group id="astroDawn" typeId="sunRange">
-				<label>Astro Dawn</label>
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.astro/ESH-INF/thing/sun.xml]: We actually have the (undocumented...) best practice to keep labels always capitalised - so this one was actually fine.

Comment (cweitkamp) [reply extensions/binding/org.eclipse.smarthome.binding.astro/ESH-INF/thing/sun.xml]: No problem. Then I will do it the other way around. Revert these changes and change the ones which do not fit to the new pattern.

Is this true for the translated labels too?

Comment (kaikreuzer) [reply extensions/binding/org.eclipse.smarthome.binding.astro/ESH-INF/thing/sun.xml]: Yes, imho makes sense to have a common way in the UI.
Should definitely also be documented, but haven't found a good place and the time yet :-/

File: extensions/binding/org.eclipse.smarthome.binding.astro/ESH-INF/i18n/astro_de.properties [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+thing-type.astro.moon.group.distance.description = Die Entfernung zum Mond
+thing-type.astro.moon.group.perigee.label = Perig�um
+thing-type.astro.moon.group.perigee.description = Das Perig�um des Mondes
+thing-type.astro.moon.group.apogee.label = Apog�um
+thing-type.astro.moon.group.apogee.description = Das Apog�um des Mondes
+thing-type.astro.moon.group.position.description = Die Position des Mondes
+
+# channel group types
+channel-group-type.astro.position.label = Position
+channel-group-type.astro.position.description = Die Position des Planeten
```
[the remark this answers is not in the document]
Comment (cweitkamp) [reply extensions/binding/org.eclipse.smarthome.binding.astro/ESH-INF/i18n/astro_de.properties]: @NorbertHD Thanks. I like "Himmelskörper" and changed it.

**Pull request merged**